extensions/azurecore/src/azureResource/commands.ts

```diff
@@ -181,17 +181,22 @@ export function registerAzureResourceCommands(appContext: AppContext, azureViewT
 		vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
 	});
 
-	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode) => {
+	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode | azdata.ObjectExplorerContext) => {
 		if (!node) {
 			return;
 		}
-
-		const treeItem: azdata.TreeItem = await node.getTreeItem();
-		if (!treeItem.payload) {
-			return;
+		let connectionProfile: azdata.IConnectionProfile = undefined;
+		if (node instanceof TreeNode) {
+			const treeItem: azdata.TreeItem = await node.getTreeItem();
+			if (!treeItem.payload) {
+				return;
+			}
+			// Ensure connection is saved to the Connections list, then open connection dialog
+			connectionProfile = Object.assign({}, treeItem.payload, { saveProfile: true });
+		} else if (node.isConnectionNode) {
```

Comment:
Is checking for isConnectionNode actually necessary? Seems like we could just assign connectionProfile and then right after this if/else if that's undefined just return (which I wouldn't expect to happen...so probably log something to console as well)

Reply:
The node only has a connectionProfile if it's a connection node

Reply:
Yes, but I'm saying that if it isn't a connection node then we can't really do anything useful anyways - and honestly that's probably a bug since I would expect this action to only show up for connection nodes.

But I guess the default action to open an empty connection dialog is fine too. I just would hope we never actually get in that scenario since that's not really the point of this action and likely indicates a bug elsewhere.